.github/actions/artifacts/action.yml

```diff
@@ -20,13 +24,22 @@ inputs:
 runs:
   using: 'composite'
   steps:
-    - name: Upload to codecov
-      uses: codecov/codecov-action@e0b68c6749509c5f83f984dd99a76a1c1a231044 # v4.0.1
-      with:
-        token: ${{ inputs.token }}
-        flags: ${{ inputs.type }}
-        files: ${{ inputs.files }}
-        override_commit: ${{ inputs.commit_sha}}
-        plugin: noop
-        verbose: true
+    - name: Download and Verify Codecov CLI
+      shell: bash
+      run: |
+        curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import # One-time step
+        curl -Os https://cli.codecov.io/latest/linux/codecov
+        curl -Os https://cli.codecov.io/latest/linux/codecov.SHA256SUM
+        curl -Os https://cli.codecov.io/latest/linux/codecov.SHA256SUM.sig
+        gpgv codecov.SHA256SUM.sig codecov.SHA256SUM
+        shasum -a 256 -c codecov.SHA256SUM
+        chmod +x codecov
+    - name: Upload Coverage
+      shell: bash
       continue-on-error: true
+      run: |
+        files=$(echo ${{ inputs.files }} | sed 's/^/--file /;s/,/ --file /')
```

Comment:
This is fraught, whenever the variable is likely to contain whitespace, quotes, other shell characters (and this is). The `${{` syntax will textually substitute the filenames so all characters in filesnames become part of the _source_ of the shell script -- an escaping nightmare.

Instead, pass all values from GHA to sh as environment variables. This looks similar but behaves very differently as the shell now has no confusion about which characters are the shell script and which are parts of the parameters.

```bash
env:
  INPUT_FILES: ${{ inputs.test_result_files }}
```